vmaas/load.go

```diff
@@ -1086,3 +1088,88 @@ func loadCpeID2Label(c *Cache) {
 	}
 	c.CpeID2Label = ret
 }
+
+func loadCSAFProductStatus(c *Cache) {
+	defer utils.TimeTrack(time.Now(), "CSAF product status")
+
+	cnt := getCount("csaf_product_status", "*")
+	rows := getAllRows("csaf_product_status", "id,name")
+
+	cache := make(map[int]string, cnt)
+
+	type ProductStatusRow struct {
+		ID    int
+		label string
+	}
+	psr := ProductStatusRow{}
+
+	for rows.Next() {
+		if err := rows.Scan(&psr.ID, &psr.label); err != nil {
+			panic(fmt.Errorf("failed to scan csaf_product_status row: %s", err.Error()))
+		}
+		cache[psr.ID] = psr.label
+	}
+
+	c.CSAFProductStatus = cache
+}
+
+func loadCSAFCVE(c *Cache) {
+	defer utils.TimeTrack(time.Now(), "CSAF CVEs")
+
+	cnt := getCount("csaf_cve_product", "*")
+	rows := getAllRows("csaf_cve_product", "id,cve_id,csaf_product_id,csaf_product_status_id")
+
+	csafCVEProduct := make(map[CSAFProductID][]CSAFCVEProduct, cnt)
+	ccp := CSAFCVEProduct{}
+
+	for rows.Next() {
+		if err := rows.Scan(&ccp.ID, &ccp.CVEID, &ccp.CSAFProductID, &ccp.CSAFProductStatusID); err != nil {
+			panic(fmt.Errorf("failed to scan csaf_cve_product row: %s", err.Error()))
+		}
+		csafCVEProduct[ccp.CSAFProductID] = append(csafCVEProduct[ccp.CSAFProductID], ccp)
+	}
+
+	rows = getAllRows("csaf_product", "id,cpe_id,package_name_id,package_id,module_stream")
+	cnt = getCount("csaf_product", "*")
+
+	cveCache := make(map[CSAFProduct]CSAFCVEs, cnt)
+
+	type CSAFProductRow struct {
+		ID      int
+		Product CSAFProduct
+	}
+
+	cpr := CSAFProductRow{}
+	for rows.Next() {
+		if err := rows.Scan(&cpr.ID,
+			&cpr.Product.CpeID,
+			&cpr.Product.PackageNameID,
+			&cpr.Product.PackageID,
+			&cpr.Product.ModuleStream); err != nil {
+			panic(fmt.Errorf("failed to scan csaf_product row: %s", err.Error()))
+		}
+
+		cveProducts := csafCVEProduct[CSAFProductID(cpr.ID)]
+		cves := CSAFCVEs{
+			Fixed:   make([]CVEID, 0),
+			Unfixed: make([]CVEID, 0),
+		}
+
+		for _, cveProduct := range cveProducts {
+			if c.CSAFProductStatus[cveProduct.CSAFProductStatusID] == "fixed" {
```

Comment:
```
1	first_affected
2	first_fixed
3	fixed
4	known_affected
5	known_not_affected
6	last_affected
7	recommended
8	under_investigation
```
I'd like to ask here, which IDs we should put into fixed, which into unfixed and if some we should skip?

Reply:
fixed: 2+3
unfixed: all rest
skip: nothing
What do you think?

Reply:
I'm not sure to be honest, I'd do the following:
fixed: 3
unfixed: 4
skip: everything else

I'd need to check CSAF docs but as far as I remember `first_affected` and `first_fixed` contain only the first package version which had CVE which is not important for us. I think `first_affected` could be even tied to a CVE which is fixed, it just shows which version was first affected by the CVE. I agree with skipping `known_not_affected`, maybe it can be used in the far future to narrow down false positives but definitely not now. `last_affected` could be also a fixed CVE, it shows last version which is affected by the CVE. `recommended` is the recommended version which to install to fix the issue, so it is part of `fixed` but I'd skip that since we are not interested in knowing which update to apply. `under_investigation` does not have CVE assigned so for us it is skip. Take this explanation with a grain of salt since it's been a while since I looked at CSAF content.
I'm afraid we will know more once we start testing it and comparing results to OVAL.